src/utils/src/bytes.rs

```diff
@@ -1,12 +1,18 @@
 // Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
-use std::convert::TryInto;
+pub fn read_le_u16(input: &[u8]) -> u16 {
+    let mut array = [0u8; 2];
+    for (byte, read) in array.iter_mut().zip(input.iter().cloned()) {
+        *byte = read as u8;
+    }
+    u16::from_le_bytes(array)
+}
 
-pub fn read_le_u16(input: &mut &[u8]) -> u16 {
-    let (int_bytes, rest) = input.split_at(std::mem::size_of::<u16>());
-    *input = rest;
-    u16::from_le_bytes(int_bytes.try_into().unwrap())
+pub fn write_le_u16(buf: &mut [u8], n: u16) {
+    for (byte, read) in buf.iter_mut().zip(u16::to_le_bytes(n).iter().cloned()) {
+        *byte = read as u8;
+    }
 }
 
 pub fn read_le_u32(input: &[u8]) -> u32 {
```

Comment:
Can this also be used in `get_klapic_reg` ?

Reply:
Both `get_klapic_reg` and `set_klapic_reg` deal with signed values ([i8] instead of [u8]) and the current utility functions only handle unsigned. However, this appears to be the only place where signed values are read and written, and I was unsure of how to address this. Should we write more functions to deal specifically with signed values?

Reply:
I see. It would be nice to have methods for signed values as well. At least for what we need in the case of `get_klapic_reg` and `set_klapic_reg`.

Also, I just realized that these methods are somewhat redundant. I think we could have a macro that auto generates them. You can take a look at [vm-memory - endian_type](https://github.com/rust-vmm/vm-memory/blob/bb29ec8713f9a798557abdd7e1449ec065cfc095/src/endian.rs#L43) as an example.